packages/mcl/src/src/mcl/commands/compare_disko.d

```diff
@@ -0,0 +1,188 @@
+
+    foreach (string m; machines){
+        MachineChanges mc;
+        mc.machine = m;
+        if (m in machineOptionsRootOld && !(m in machineOptionsRootNew)){
+            static foreach (setting; diskoOptions){
+                __traits(getMember, mc, setting) = ChangeStatus.Removed;
+            }
+        }
+        else if (m in machineOptionsRootNew && !(m in machineOptionsRootOld)){
+            static foreach (setting; diskoOptions){
+                __traits(getMember, mc, setting) = ChangeStatus.New;
+            }
+        }
+        else{
+            static foreach (setting; diskoOptions){
+                if (machineOptionsRootOld[m][setting] == machineOptionsRootNew[m][setting]){
+                    __traits(getMember, mc, setting) = ChangeStatus.Unchanged;
+                }
+                else{
+                    __traits(getMember, mc, setting) = ChangeStatus.Changed;
+                }
+            }
+        }
+        machineChanges ~= mc;
+    }
+
+    create_comment(machineChanges);
+
+    execute(["git", "worktree" , "remove", worktreeBaseBranch, "--force"]);
+}
+
+void create_comment(MachineChanges[] machineChanges){
+    string data = "Thanks for your Pull Request!";
+    if (machineChanges.length == 0){
+        data ~="\n\n✅  There have been no changes to disko configs";
+    }
+    else{
+        data ~= "\n\nBellow you will find a summary of machines and whether their disko attributes have differences.";
+        data ~= "\n\n**Legend:**";
+        data ~= "\n🟩 = No changes";
+        data ~= "\n⚡ = Something is different";
+        data ~= "\n🗑 = Has been removed";
+        data ~= "\n🧩 = Has been added";
+
+        data ~= "\n\n";
+        foreach (string field; __traits(allMembers, MachineChanges)){
+            data~="| " ~ field ~ " ";
+        }
+        data ~= "|\n";
+        foreach (string field; __traits(allMembers, MachineChanges)){
+            data~="| --- ";
+        }
+        data ~= "|\n";
+
+        foreach(mc; machineChanges){
+            foreach (string field; __traits(allMembers, MachineChanges)){
+                static if (is(typeof(__traits(getMember, mc, field)) == string)){
+                    data~="| " ~ __traits(getMember, mc, field) ~ " ";
+                }
+                else {
+                    data~="| " ~ statusToSymbol(__traits(getMember, mc, field)) ~ " ";
+                }
+            }
+            data ~= "|\n";
+        }
+    }
+    write("comment.md", data);
+}
+
+
+string constructCommandAttr(string flakePath, string[] machines){
+    string attr = "let flake = (builtins.getFlake (builtins.toString " ~ flakePath ~ ")); in { ";
+    foreach (m; machines){
+        attr ~= m ~ " = { ";
+        foreach (option; diskoOptions){
+            attr ~= option ~ " = flake.nixosConfigurations." ~ m ~ ".config.disko.devices." ~ option ~ "; ";
+        }
+        attr ~= "}; ";
+    }
+    attr ~= "}";
+    return attr;
+}
```

Comment:
Consider using appender!string instead of "~", as it's faster and more efficient.
https://forum.dlang.org/thread/ppbczcbqhaqalgwcjbba@forum.dlang.org